src/main/java/ch/njol/skript/effects/EffScriptFile.java

```diff
@@ -27,90 +9,136 @@
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Effect;
 import ch.njol.skript.lang.Expression;
+import ch.njol.skript.registrations.Feature;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
 import org.skriptlang.skript.lang.script.Script;
 import ch.njol.skript.lang.SkriptParser;
 import ch.njol.skript.util.FileUtils;
 import ch.njol.util.Kleenean;
 import ch.njol.util.OpenCloseable;
 import org.bukkit.event.Event;
-import org.jetbrains.annotations.Nullable;
 
 import java.io.File;
+import java.io.FileFilter;
 import java.io.IOException;
 import java.util.Set;
 
-@Name("Enable/Disable/Reload Script File")
-@Description("Enables, disables, or reloads a script file.")
+@Name("Enable/Disable/Unload/Reload Script")
+@Description("""
+	Enables, disables, unloads, or reloads a script.
+	
+	Disabling a script unloads it and prepends - to its name so it will not be loaded the next time the server restarts.
+	If the script reflection experiment is enabled: unloading a script terminates it and removes it from memory, but does not alter the file.""")
 @Examples({
 	"reload script \"test\"",
 	"enable script file \"testing\"",
-	"unload script file \"script.sk\""
+	"unload script file \"script.sk\"",
+	"set {_script} to the script \"MyScript.sk\"",
+	"reload {_script}"
 })
-@Since("2.4")
+@Since("2.4, INSERT VERSION (unloading)")
 public class EffScriptFile extends Effect {
 
 	static {
 		Skript.registerEffect(EffScriptFile.class,
-			"(1:(enable|load)|2:reload|3:(disable|unload)) s(c|k)ript [file] %string%"
+			"(1:(enable|load)|2:reload|3:disable|4:unload) script [file|named] %string%",
+			"(1:(enable|load)|2:reload|3:disable|4:unload) skript file %string%",
+			"(1:(enable|load)|2:reload|3:disable|4:unload) %scripts%"
 		);
+		/*
+			The string-pattern must come first (since otherwise `script X` would match the expression)
+			and we cannot get a script object for a non-loaded script.
+		 */
 	}
-	
-	private static final int ENABLE = 1, RELOAD = 2, DISABLE = 3;
-	
+
+	private static final int ENABLE = 1, RELOAD = 2, DISABLE = 3, UNLOAD = 4;
+
 	private int mark;
 
-	@SuppressWarnings("NotNullFieldNotInitialized")
-	private Expression<String> fileName;
+	private @UnknownNullability Expression<String> stringExpression;
+	private @UnknownNullability Expression<Script> scriptExpression;
```

Comment:
Not resolved

Reply:
Actually there's already a scripts field and I can't call it script either because the variable is called that, so I've changed the string one but left the other.